test/command.js

```diff
@@ -165,9 +165,29 @@ test('$ handles invalid escape sequence', async t => {
 	t.is(stdout, '\\u');
 });
 
-test('$ coerces all interpolated expressions to strings', async t => {
-	const {stdout} = await $`node test/fixtures/echo.js ${0} ${undefined} ${null}`;
-	t.is(stdout, '0\nundefined\nnull');
+test('$ throws on invalid expression in interpolation', async t => {
```

Comment:
We might want to break this into multiple tests.
[Ava macros](https://github.com/avajs/ava/blob/1c567642a31d4bc24c4421ba520b02e25ce7363a/docs/01-writing-tests.md#retrieving-test-metadata) can be used for this purpose too, like we do in https://github.com/sindresorhus/execa/blob/main/test/stdio.js

Reply:
TIL! That cleaned up a ton of duplication and encouraged me to add a few more test cases: https://github.com/sindresorhus/execa/pull/528/commits/602fd8819a3dc425fe633c9405af98ac40e600e3

Reply:
Here's a preview of the test titles:
<img width="1278" alt="Screenshot 2023-03-04 at 9 28 22 PM" src="https://user-images.githubusercontent.com/32409546/222943522-c843f88a-46b7-4e2c-8a61-5817192b239d.png">